src/main/java/dev/blachut/svelte/lang/Svelte.flex

```diff
@@ -43,36 +47,44 @@ WHITE_SPACE=\s+
   "<!--"                  { yybegin(VERBATIM_COMMENT); return HTML_FRAGMENT; }
   "<script" | "<style"    { yybegin(VERBATIM_HTML); return HTML_FRAGMENT; }
   "<"                     { yybegin(HTML_TAG); return HTML_FRAGMENT; }
-  "{" / \s*[#:/]          { yybegin(SVELTE_TAG); return START_MUSTACHE; }
+  "{"\s*"#"               { yybegin(SVELTE_TAG_PRE); return START_OPENING_MUSTACHE; }
+  "{"\s*":"               { yybegin(SVELTE_TAG_PRE); return START_INNER_MUSTACHE; }
+  "{"\s*"/"               { yybegin(SVELTE_TAG_PRE); return START_CLOSING_MUSTACHE; }
   "{"                     { yybegin(SVELTE_INTERPOLATION); return START_MUSTACHE; }
   {WHITE_SPACE}           { return WHITE_SPACE; }
 }
 
+<SVELTE_TAG_PRE> {
+  "if"               { yybegin(SVELTE_TAG); return IF; }
+  "each"             { yybegin(SVELTE_TAG); return EACH; }
+  "await"            { yybegin(SVELTE_TAG); return AWAIT; }
+  "then"             { yybegin(SVELTE_TAG); return THEN; }
+  "catch"            { yybegin(SVELTE_TAG); return CATCH; }
+  "else"             { yybegin(SVELTE_TAG); return ELSE; }
+  {ID}               { yybegin(SVELTE_TAG); return BAD_CHARACTER; }
+  {WHITE_SPACE}      { return BAD_CHARACTER; }
+}
 <SVELTE_TAG, SVELTE_TAG_PAREN_AWARE> {
-  "#if"              { return IF; }
-  "if"               { return ELSE_IF; }
-  "/if"              { return END_IF; }
-
-  "#each"            { return EACH; }
+  "if"               { return IF; }
```

Comment:
@tomblachut - not actually sure why we need "if" and "then" in this state. It will try to parse incorrectly in such example: `<#each life as ...>`, because `if` is part of `life`. And independent "if", "then" should not happen after `<#each`, `<#if`, `<#else`, etc,  Also it makes sense to change `as` to `\s+"as"\s+` otherwise it is impossible to use "as" as part of variable name: `<#each asset as ...>`.

Reply:
You make fair points. Reason is that we need to support `else if` which in ideal implementation  must be 2 separate tokens because there's a whitespace between them. 

IDE could break slightly when you ignore that and I don't really know how everything interacts together so I wanted to keep safe.

For the same reason `\s+"as"\s+` is not ideal. As you can see I broke that rule in `"{"\s*"#"` but couldn't do it better after many hours spent. I was getting incomprehensible edge case bugs.

Reply:
Here is a commit which has a correct (I hope) `else if` parsing, and which returns tokens without spaces: https://github.com/tiatin/svelte-intellij/commit/49b45ddbad47b64162426855a1b53a256c8c6bf8. You can use it as a base to fix your PR. You can test it using this build https://github.com/tiatin/svelte-intellij/releases/tag/v0.7.4.

I also don't understand why you use `"{"\s*"#"` instead of ` "{#"`, as this token should not contain any spaces in between.

Reply:
Actually Svelte allows spaces there, which is weird IMO but that's how it is. On the other hand you can't put space between # and a word.

I'm going to try your build, it looks promising! Would you care to open PR? I'd like to avoid stealing your attribution

Reply:
This edge case still breaks.

![image](https://user-images.githubusercontent.com/615013/61581240-2fbf2200-ab1c-11e9-8cdf-585d3c0da486.png)

https://svelte.dev/repl/79348a7417184f2db9668802f2530401?version=3.6.7

Reply:
I agree with your case: `as` is actually not reserved, so it can be a valid variable. Basically need to add one more flag into lexer to track that there was some expression before detecting a real `as`. I can update lexer on Monday and will let you know. I actually don't mind if you "steal" the work, so you can copy lexer commits into your `feature/parser-error-messages` branch and close this PR sooner.

Reply:
Cool! I'll see about that. Have a nice weekend :)

Reply:
Have a nice weekend! Thank you for the plugin! Hard to code in svelte without it.

Reply:
Thx for kind words. Okay I've handled as issue totally differently. `each as as as` and `await then then then` is still unsupported, but lexer file is already getting incomprehensible and we aren't even close to "fun" parts like passing object literals containing as variable to functions inside iterable part of `each`. 

```
{#each foo({x: as}) as a, i (a)}
```

Whole thing needs to be redesigned

Reply:
Sounds fair. Thank you!